foxglove_bridge_base/include/foxglove_bridge/websocket_server.hpp

```diff
@@ -906,6 +909,22 @@ inline void Server<ServerConfiguration>::handleTextMessage(ConnHandle hdl, Messa
                             "Client was not subscribed to connection graph updates");
       }
     } break;
+    case FETCH_ASSET: {
+      if (!_handlers.fetchAssetHandler) {
+        return;
+      }
+
+      const auto uri = payload.at("uri").get<std::string>();
+      const auto requestId = payload.at("requestId").get<uint32_t>();
```

Comment:
what happens if these are missing or the wrong type? if an exception is thrown here, will it be caught somewhere?

Reply:
Yes, any exception that is not handled in the operation handler would be caught here:
https://github.com/foxglove/ros-foxglove-bridge/blob/9a206a06a2e8fd1564f066840a2b7e86d2cf74b7/foxglove_bridge_base/include/foxglove_bridge/websocket_server.hpp#L560-L571